SharedContent/cs-winui-template/WinUI3TemplateCs/MainWindow.xaml.cs

```diff
@@ -0,0 +1,110 @@
+        {
+            this.InitializeComponent();
+
+            Title = Settings.FeatureName;
+
+            HWND hwnd = (HWND)WinRT.Interop.WindowNative.GetWindowHandle(this);
+            LoadIcon(hwnd,"Assets/windows-sdk.ico");
+            SetWindowSize(hwnd, 1220, 1080);
+            PlacementCenterWindowInMonitorWin32(hwnd);
+        }
+
+        private unsafe void LoadIcon(HWND hwnd, string iconName)
+        {
+            const int ICON_SMALL = 0;
+            const int ICON_BIG = 1;
+
+            fixed (char* nameLocal = iconName)
+            {
+                HINSTANCE hInstLocal = default(HINSTANCE);
+                HANDLE __result = LoadImage(hInstLocal, nameLocal,
+                    GDI_IMAGE_TYPE.IMAGE_ICON,
+                    GetSystemMetrics(SYSTEM_METRICS_INDEX.SM_CXSMICON),
+                    GetSystemMetrics(SYSTEM_METRICS_INDEX.SM_CYSMICON),
+                    IMAGE_FLAGS.LR_LOADFROMFILE);
+                SendMessage(hwnd, WM_SETICON, ICON_SMALL, __result.Value);
+            }
+
+            fixed (char* nameLocal = iconName)
+            {
+                HINSTANCE hInstLocal = default(HINSTANCE);
+                HANDLE __result = LoadImage(hInstLocal, nameLocal,
+                    GDI_IMAGE_TYPE.IMAGE_ICON,
+                    GetSystemMetrics(SYSTEM_METRICS_INDEX.SM_CXSMICON),
+                    GetSystemMetrics(SYSTEM_METRICS_INDEX.SM_CYSMICON),
+                    IMAGE_FLAGS.LR_LOADFROMFILE);
+                SendMessage(hwnd, WM_SETICON, ICON_BIG, __result.Value);
+            }
+        }
+
+        private void SetWindowSize(HWND hwnd, int width, int height)
+        {
+            // Win32 uses pixels and WinUI 3 uses effective pixels, so you should apply the DPI scale factor
+            var dpi = GetDpiForWindow(hwnd);
+            float scalingFactor = (float)(dpi / 96);
+            width = (int)(width * scalingFactor);
+            height = (int)(height * scalingFactor);
+
+            SetWindowPos(hwnd, HWND_TOP, 0, 0, width, height, SET_WINDOW_POS_FLAGS.SWP_NOMOVE);
+        }
+
+        private void PlacementCenterWindowInMonitorWin32(HWND hwnd)
+        {
+            RECT rc;
+            GetWindowRect(hwnd, out rc);
+            ClipOrCenterRectToMonitorWin32(rc, true, true);
+            SetWindowPos(hwnd, HWND_TOP, rc.left, rc.top, 0, 0,
+                         SET_WINDOW_POS_FLAGS.SWP_NOSIZE | 
+                         SET_WINDOW_POS_FLAGS.SWP_NOZORDER | 
+                         SET_WINDOW_POS_FLAGS.SWP_NOACTIVATE);     
+        }
+
+        private void ClipOrCenterRectToMonitorWin32(RECT prc, bool UseWorkArea, bool IsCenter)
+        {
+            HMONITOR hMonitor;
+            RECT rc;
+            int w = prc.right - prc.left;
+            int h = prc.bottom - prc.top;
+
+            hMonitor = MonitorFromRect(prc, MONITOR_FROM_FLAGS.MONITOR_DEFAULTTONEAREST);
+            MONITORINFO mi = new MONITORINFO();
+            mi.cbSize = (uint)Marshal.SizeOf(mi);
```

Comment:
Use `sizeof(MONITORINFO)` instead of `Marshal.SizeOf` if C# will let you. It results in a constant instead of a method call.

Reply:
This results in the error: 'MONITORINFO' does not have a predefined size, therefore sizeof can only be used in an unsafe context